src/main/java/com/solace/connector/kafka/connect/sink/SolaceSinkSender.java

```diff
@@ -203,7 +203,28 @@ public void sendRecord(SinkRecord record) {
       sinkTask.flush(offsets);
     }
   }
-  
+
+  /**
+   * Visible for testing.
+   */
+  void mayEnrichUserPropertiesWithKafkaRecordHeaders(final SinkRecord record,
+                                                     final BytesXMLMessage message) {
+    if (sconfig.isEmitKafkaRecordHeadersEnabled() && !record.headers().isEmpty()) {
+      final SDTMap userMap = Optional
+              .ofNullable(message.getProperties())
+              .orElseGet(JCSMPFactory.onlyInstance()::createMap);
+      record.headers().forEach(header -> {
+        try {
+          userMap.putObject(header.key(), header.value());
```

Comment:
Kafka messages can contain multiple headers under the same key. If we directly write them as Solace message properties, they'll just end up overwriting each other.

# Alternative Solution

For each Kafka header key, assign it an `SDTStream` which will contain all its values.

While this solution will resolve the multiple-header-values-under-1-key issue, it will increase complexity for when a header key only has a single value. So I'm a little on the fence about whether or not this is worth doing.

@mackenza could I also get your thoughts on this?